src/ray/util/signal_handler.cc

```diff
@@ -0,0 +1,72 @@
+#include "ray/util/signal_handler.h"
+#include "ray/util/logging.h"
+
+#include <signal.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sstream>
+
+using namespace ray;
+
+// Normally stop ray will also send the signal.
+int SignalHandlers::terminate_logging_level_ = RAY_INFO;
+// The list of signals that has installed handlers.
+std::vector<int> SignalHandlers::installed_signals_;
+// The current app name.
+std::string SignalHandlers::app_name_;
+
+void SignalHandlers::InstallSignalHandlerHelper(int sig, void (*handler)(int)) {
+  signal(sig, handler);
```

Comment:
I think [sigaction](http://man7.org/linux/man-pages/man2/sigaction.2.html) is a preferred way to install signal handlers these days.